flipt-python/flipt/authentication/__init__.py

```diff
@@ -17,3 +23,46 @@ def __init__(self, token: str) -> None:
 
     def authenticate(self, headers: dict[str, str]) -> None:
         headers["Authorization"] = f"JWT {self.token}"
+
+
+class KubernetesAuthentication(AuthenticationStrategy):
+    default_service_token_path = "/var/run/secrets/kubernetes.io/serviceaccount/token"  # noqa: S105
+
+    def __init__(self, token: str, service_account_token_path: str = default_service_token_path) -> None:
+        self.token = token
+        self.service_account_token_path = service_account_token_path
+        self.token_expiry = None
+
+    def authenticate(self, headers: dict[str, str]) -> None:
+        # Check if token is available and not expired.
+        if self.token and self.token_expiry and self.token_expiry > datetime.now(tz=datetime.timezone.utc).timestamp():
+            headers["Authorization"] = f"Bearer {self.token}"
+            return
+
+        # Read service account info from the local file system.
+        try:
+            with Path.open(Path(self.service_account_token_path), "r") as token_file:
+                service_account_token = token_file.read().strip()
+        except IOError as e:
+            raise RuntimeError("Failed to read service account token.") from e
+
+        # Send the token to flipt"s auth endpoint and collect the response.
+        payload = {"service_account_token": service_account_token}
+        try:
+            response = requests.post(
+                "http://flipt:8080/auth/v1/method/kubernetes/serviceaccount",
```

Comment:
is it possible that people may have some other hostname to access the Flipt in their cluster? @GeorgeMac

Reply:
Good spot! This has been hardcoded to pass the test haha

Reply:
See my suggestion above to pass in the URL as an argument to the Kuberentes client @Dents6679. Then we can thread the URL in here before we make the request.

```suggestion
                f"http://{self.url}/v1/method/kubernetes/serviceaccount",
```